rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,771 @@
+
+```jsx
+import { makeStyles, makeResetStyles, mergeClasses, shorthands } from '@griffel/react';
+
+const useBaseClassname = makeResetStyles({
+  ':hover': {
+    ...shorthands.padding('4px'),
+    /* other styles */
+  },
+  '::before': {
+    display: 'block',
+    content: "' '",
+  },
+  '@media (forced-colors: active)': {
+    ...shorthands.padding('4px'),
+    /* other styles */
+  },
+});
+// ⬇️⬇️⬇️
+// produces 1 class
+
+const useClasses = makeStyles({
+  circular: {
+    ...shorthands.borderRadius('10px'),
+  },
+  primary: {
+    color: 'pink',
+  },
+});
+// ⬇️⬇️⬇️
+// produces 4/1 classes to be conditionally applied
+
+function Component(props) {
+  const baseClassName = useBaseClassname();
+  const classes = useClasses();
+
+  const className = mergeClasses(baseClassName, props.circular && classes.circular, props.primary && classes.primary);
+
+  return <div className={className} />;
+}
+```
+
+# Advanced
+
+## Understanding selectors complexity
+
+CSS Selectors are matched by browser engines from [right to left (bottom-up parsing)][stackoverflow-selectors-match]:
+
+```mermaid
+flowchart RL
+    C[li] --> B[ul] --> A[.menu]
+```
+
+```css
+.menu ul li {
+  color: #00f;
+}
+```
+
+> The browser first checks for `li`, then `ul`, and then `.menu`.
+
+It means that we need to understand selectors that we are writing and avoid _wide_ selectors. A component below will be used as an example:
+
+```jsx
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  test: {
+    color: 'orange',
+
+    '> *': {
+      color: 'red',
+    },
+    '> h1': {
+      color: 'magenta',
+    },
+    '> div': {
+      color: 'green',
+    },
+    '> .ui-button': {
+      color: 'blue',
+    },
+  },
+});
+
+function App() {
+  const classes = useClasses();
+
+  return (
+    <>
+      <div className={classes.test}>
+        <h1>Hello World</h1>
+        <button class="ui-button">A button</button>
+      </div>
+
+      {/* Renders 500 empty divs */}
+      {Array.from({ length: 500 }, (_, i) => (
+        <div key={i} />
+      ))}
+    </>
+  );
+}
+```
+
+#### Measurements
+
+##### No selector (best)
+
+Matches only an element where is applied.
+
+- selector `.fe3e8s9`
+- match_attempts **1**
+- match_count **1**
+
+##### `> *` (worst)
+
+"\*" matches any element on a page.
+
+- selector `.fzbuleu > *`
+- match_attempts **502**
+- match_count **1**
+
+##### `> h1` & `> div` (non ideal)
+
+Targets all `h1` & `div` tags.
+
+- selector `.fohk16y > h1`
+- match_attempts **1**
+- match_count **1**
+
+In this example performs better as page has a single `h1` tag, this will not be a case in the real app that it is visible on the next example with `div`.
+
+- selector `.fq4d7o6 > div`
+- match_attempts **501**
+- match_count **1**
+
+#### Targeting a classname (better)
+
+The most effective out there as we target only elements with a specific classname. However, it will not save from the case with 1k buttons in the app.
```

Comment:
```suggestion
The most effective since we only target elements with a specific class. However, this is still not optimal if there are 1000 elements with the class `ui-button` on the page.
```